### PR TITLE
feat: XAR installation from gulp source-stream

### LINKED ISSUE #37: improving docs
There should be a bit more introductory text for those not being so familiar with gulp and npm.

* how to install it
* how to update it

### LINKED ISSUE #124: Add task to install a XAR package
`install` or `installPackage` that can be used in `.pipe()`

**Prerequisite**

- https://github.com/eXist-db/node-exist/pull/129 is merged into node-exist

**Example**

```js
const { src } = require('gulp')
const { createClient } = require('@existdb/gulp-exist')
const existClient = createClient()

function install () {
  return src('my-app.xar').pipe(existClient.install())
}
exports.install = install
```

### PR DESCRIPTION
**Pull request by line-o**
closes #126
closes #124 
closes #37 

BREAKING CHANGE: The next version of gulp-exist will no longer be fully compatible with exist versions 2.2 and 3.6.1

- createClient returns new method `install`
    - If provided with the unique package identifier of a xar it allows
      to install a XAR package in the connected existdb.
      The contents and filename are read from a VinylFileStream (gulp.src).
- add type definitions to all user facing functions.
  This will give fair autocompletion and
  hints in  IDEs that read this information (eg. VSCode).

- reorganize index.js
  - move and group exports at the end
  - bind gulp-exist methods to XMLRPCClient. This  gets rid
    of one wrapping function each.

- update to @existdb/node-exist 3.0.1

- add small package test-app.xar for testing
- add tests for package installation

- adapt example gulpfile.js
  - require local gulp-exist to see hints in action
  - use install method in example `installXar` task
- add install section with example to README.MD

### Example Installation

```js
 const { src } = require('gulp')
 const { createClient } = require('@existdb/gulp-exist')

 // override defaults
 const exist = createClient({
     basic_auth: {
         user: 'admin',
         pass: ''
     }
 })
 // this MUST be the unique package identifier of the XAR you want to install 
 const packageUri = 'http://exist-db.org/apps/test-app'

 function install () {
     return src('spec/files/test-app.xar')
         .pipe(exist.install({ packageUri }))
 }
 ```

### CODE REVIEW

**Comment by line-o** [comment]
What do you think @duncdrum ?

File: spec/examples/gulpfile.js [hunk 1]
```diff
@@ -1,22 +1,23 @@
 /**
  * an example gulpfile to make ant-less existdb package builds a reality
  */
-const fs = require('fs')
 const { src, dest, watch, series, parallel, lastRun } = require('gulp')
-const { createClient } = require('@existdb/gulp-exist')
-const { connect } = require('@existdb/node-exist')
+const { createClient } = require('../../index')
```
Comment (duncdrum) [inline spec/examples/gulpfile.js]: Should this not be `require(‘@existdb/gulp-exist’)` in the example file, as it’s what users are likely to do?

**Review by duncdrum — COMMENTED** [review]
Two instances where I thought Switches would be more legible but not a showstopper. One question about the example file. Otherwise looks great.

File: spec/examples/gulpfile.js (hunk 1 above)
Comment (line-o) [reply spec/examples/gulpfile.js]: Yeah, you are probably right. I changed it to be able to see the code hints.

**Comment by line-o** [comment]
@duncdrum where should we switch to `switch`?

**Review by duncdrum — APPROVED** [review]

**Pull request merged**